src/Umbraco.Core/Events/EventAggregator.cs

```diff
@@ -41,6 +41,20 @@ public Task PublishAsync<TNotification>(TNotification notification, Cancellation
             PublishNotification(notification);
             return PublishNotificationAsync(notification, cancellationToken);
         }
+
+        /// <inheritdoc/>
+        public void Publish<TNotification>(TNotification notification)
+            where TNotification : INotification
+        {
+            // TODO: Introduce codegen efficient Guard classes to reduce noise.
+            if (notification == null)
+            {
+                throw new ArgumentNullException(nameof(notification));
+            }
+
+            PublishNotification(notification);
+            Task.WaitAll(PublishNotificationAsync(notification));
```

Comment:
I think (hope) in our case this will be ok but Task.WaitAll can also deadlock if called from a UI thread or a threadpool thread (which will most likely be us) if the threadpool is exhausted (but in that case i think we're screwed anyways) or also if this is called from an async method in a threadpool thread (which we'd hope isn't done). 

I think we're ok to leave this but make sure we don't use this one when already in async contetx.